Update C API test data comparison for s390x

shahidhs-ibm opened the pull request:
Base64 encode/decode is not endian-dependent. The hash passed to Base64 will generate different string on big-endian platform. Correcting the test data comparison for s390x architecture.

shahidhs-ibm committed:
Correct the AppendHash test data comparison for s390x

shahidhs-ibm committed:
Correct _USE_C_API test data comparison for s390x

shahidhs-ibm committed:
Correcting the Python coding style (pylint)

Correcting the following comment issued by pylint:
C: 25, 0: standard import "import platform" should be placed before "import numpy as np" (wrong-import-order)

shahidhs-ibm committed:
Correcting code to conform to Google C++ Style Guide

Correcting as per the changes suggested by "clang-format".

tensorflow-jenkins commented:
Can one of the admins verify this patch?

iganichev commented on tensorflow/python/framework/function_test.py:
```diff
@@ -765,8 +766,12 @@ def Foo(x, y, z):
     if ops._USE_C_API:  # pylint: disable=protected-access
-      self.assertEqual("Foo_aCYSbwBkR5A",
-                       Foo.instantiate([dtypes.float32] * 3).name)
+      if platform.machine() == "s390x":
```
Since base64 output should be the same for all big endian machines, lets make this more generic and check for " sys.byteorder == 'big' " instead.

iganichev commented on tensorflow/c/c_api_function_test.cc:
```diff
@@ -1462,7 +1462,11 @@ TEST_F(CApiFunctionTest, AppendHash) {
                  /*append_hash=*/true);
   tensorflow::FunctionDef fdef;
   ASSERT_TRUE(GetFunctionDef(func_, &fdef));
+#if ((__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__) && defined(__s390x__))
```
It seems like we can make the check more generic and remove " && defined(__s390x__)".

shahidhs-ibm committed:
Making the changes generic for all big endian architectures

Changes made as per review comment: https://github.com/tensorflow/tensorflow/pull/15390#pullrequestreview-84306138

shahidhs-ibm committed:
Making the changes generic for all big endian architectures

Changes made as per review comment: https://github.com/tensorflow/tensorflow/pull/15390#discussion_r157631673

shahidhs-ibm committed:
Correcting code to conform to Google C++ Style Guide

shahidhs-ibm commented:
@iganichev Thank you for your review comments. Code changes incorporated.

drpngx commented:
Jenkins, test this please.
